hadoop-hdfs-project/hadoop-hdfs-httpfs/src/test/java/org/apache/hadoop/fs/http/server/TestHttpFSServer.java

```diff
@@ -120,6 +122,25 @@
  */
 public class TestHttpFSServer extends HFSTestCase {
 
+  /**
+   * define metric getters for unit tests.
+   */
+  private static Callable<Long> defaultEntryMetricGetter = () -> 0L;
+  private static Callable<Long> defaultExitMetricGetter = () -> 1L;
+  private static HashMap<String, Callable<Long>> metricsGetter =
+      new HashMap<String, Callable<Long>>() {
+        {
+          put("LISTSTATUS",
+              () -> HttpFSServerWebApp.get().getMetrics().getOpsListing());
+          put("MKDIRS",
+              () -> HttpFSServerWebApp.get().getMetrics().getOpsMkdir());
+          put("GETFILESTATUS",
+              () -> HttpFSServerWebApp.get().getMetrics().getOpsStat());
+        }
+      };
+
```

Comment:
It's not clear to me why we need this metricsGetter.   Seems to add complexity with no added value.  Can't you just call the appropriate HttpFSServerWebApp.get().getMetrics().get* function wherever this is used?

Reply:
I found that the best way to check the metrics is to inject code in `getStatus`. This reduced the diff significantly. getStatus() accepts the cmd as a string, so adding if-then-else for each cmd will be more difficult to maintain. Finally, it is easier to add a command to the metricsGetter to extend the set of tests.
Also, the default callable saves NPE in case the command is not mapped to a metricGetter.

Reply:
Oh I see that makes sense for `getStatus()`.  I was looking at `testMkdirs() `when I was trying to understand why you were using it, and it didn't make sense to me.  I don't think you should use the metricsGetter in `testMkdirs()`, or at least not use `getOrDefault()` - the default case here does not make sense - it renders that part of the test meaningless.  I'd prefer to just get the ops directly in this case.

Reply:
Thanks @jbrennan333 .
This makes sense.
I fixed `testMkdirs`, `createDirWithHttp`, `testGlobFilter`, and `testHdfsAccess`